pyccl/pk2d.py

```diff
@@ -282,57 +323,44 @@ def get_spline_arrays(self):
-                             "try swapping the operands.")
-        if (self.psp.amin < other.psp.amin
-                or self.psp.amax > other.psp.amax):
-            raise ValueError("The 2nd operand has its data defined over a "
-                             "smaller a range than the 1st operand. To avoid "
-                             "extrapolation, this operation is forbidden. If "
-                             "you want to operate on the smaller support, "
-                             "try swapping the operands.")
+        if self not in other:
+            raise ValueError(
+                "The 2nd operand has its data defined over a smaller range "
+                "than the 1st operand. To avoid extrapolation, this operation "
+                "is forbidden. If you want to operate on the smaller support, "
+                "try swapping the operands.")
 
         a_arr_a, lk_arr_a, pk_arr_a = self.get_spline_arrays()
         a_arr_b, lk_arr_b, pk_arr_b = other.get_spline_arrays()
-        if not (a_arr_a.size == a_arr_b.size and lk_arr_a.size == lk_arr_b.size
+        if not (a_arr_a.size == a_arr_b.size
+                and lk_arr_a.size == lk_arr_b.size
                 and np.allclose(a_arr_a, a_arr_b)
                 and np.allclose(lk_arr_a, lk_arr_b)):
-            warnings.warn("The arrays of the two Pk2D objects are defined at "
-                          "different points in k and/or a. The second operand "
-                          "will be interpolated for the operation.\n"
-                          "The resulting Pk2D object will be defined for "
-                          f"{self.psp.lkmin} <= log k <= {self.psp.lkmax} and "
-                          f"{self.psp.amin} <= a <= {self.psp.amax}.",
-                          category=CCLWarning)
-
-            # Since the power spectrum is evalulated on a smaller support than
-            # where it was defined, no extrapolation is necessary and the
-            # dependence on the cosmology in moot.
-            # CosmologyVanillaLCDM is being imported here instead of the top of
-            # the module due to circular import issues there.
-            from .core import CosmologyVanillaLCDM
-            dummy_cosmo = CosmologyVanillaLCDM()
-            pk_arr_b = np.array([other.eval(k=np.exp(lk_arr_a),
-                                            a=a_,
-                                            cosmo=dummy_cosmo)
-                                 for a_ in a_arr_a])
+            warnings.warn(
+                "Operands defined over different ranges. "
+                "The result will be interpolated and clipped to "
+                f"{self.psp.lkmin} <= log k <= {self.psp.lkmax} and "
+                f"{self.psp.amin} <= a <= {self.psp.amax}.", CCLWarning)
+            pk_arr_b = other(np.exp(lk_arr_a), a_arr_a)
+
         return a_arr_a, lk_arr_a, pk_arr_a, pk_arr_b
 
     def __add__(self, other):
```

Comment:
Can we document that `other` can be a number?

Reply:
done in the class docstring